src/source/worker.js

```diff
@@ -8,12 +9,30 @@ const assert = require('assert');
 
 const globalRTLTextPlugin = require('./rtl_text_plugin');
 
+import type {
+    WorkerSource,
+    WorkerTileParameters,
+    WorkerTileCallback,
+    TileParameters,
+    RedoPlacementParameters,
+    RedoPlacementCallback
+} from '../source/source';
+
 /**
  * @private
  */
 class Worker {
-    constructor(self) {
-        this.self = self;
+    self: WorkerGlobalScope & {
+        registerWorkerSource: (string, Class<WorkerSource>) => void,
+        registerRTLTextPlugin: (any) => void
+    };
+    actor: Actor;
+    layerIndexes: { [string]: StyleLayerIndex };
+    workerSourceTypes: { [string]: Class<WorkerSource> };
+    workerSources: { [string]: { [string]: WorkerSource } };
+
+    constructor(self: WorkerGlobalScope) {
+        this.self = (self: any);
```

Comment:
👍  I don't see a cleaner way to do this.  Could we stick a `//` comment here explaining the cast to `any`?